content/en/blog/_posts/2021-09-06-memory-qos-cgroups-v2/index.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: blog
+title: 'Alpha in Kubernetes v1.22: Quality-of-Service for Memory Resources'
+date: 2021-09-01
+slug: qos-memory-resources
+---
+
+**Authors:** Tim Xu (Tencent Cloud)]
+
+Kubernetes v1.22 introduced a new alpha feature that improves how Linux nodes implement memory resource requests and limits.
+
+In prior releases, Kubernetes did not support memory quality guarantees. In current Kubernetes design, user sets requests and limits to describe  resources required for containers in pod, and Kubernetes uses these values for pod scheduling and creation. For resources such as CPU and memory, Kubernetes allocates and limits these through cgroup on Linux nodes for isolation purposes. CPU is well protected in existing implementations, but memory is not as well protected due to cgroup v1 limitations. Fortunately, cgroup v2 brings a new design and implementation to achieve full protection on memory. The new feature relies on cgroups v2 which most current operating system releases for Linux already provide. With this experimental feature, [quality-of-service for pods and containers](/docs/tasks/configure-pod-container/quality-service-pod/) extends to cover not just CPU time but memory as well.
+
+## How does it work?
```

Comment:
I wonder if it would help if we added a small explanation of how memory protection worked before this change (in cgroupv1 world).

i.e. that on container resources spec, memory.requests would only be used for scheduler and memory.limit would be enforced by underlying cgroup. memory.requests was "ignored" during the actual runtime of a workload before the introduction of this KEP. It may help explain to users significance of this feature.